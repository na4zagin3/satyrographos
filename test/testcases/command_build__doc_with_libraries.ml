module StdList = List

open Satyrographos_testlib
open TestLib

open Shexp_process

let satyristes =
{|
(lang "0.0.3")
(library
  (name "grcnum")
  (version "0.2")
  (sources
    ((package "grcnum.satyh" "./grcnum.satyh")
     (font "grcnum-font.ttf" "./font.ttf" ())
     (hash "fonts.satysfi-hash" "./fonts.satysfi-hash")
     ; (file "doc/grcnum.md" "README.md")
    ))
  (opam "satysfi-grcnum.opam")
  (dependencies (fonts-theano)))

(doc
  (name "example-doc")
  (build
    ((satysfi "doc-example.saty" "-o" "doc-example-ja.pdf")
     (make "build-doc")))
  (dependencies (grcnum fonts-theano)))
|}

let satysfi_grcnum_opam =
  "satysfi-grcnum.opam", TestLib.opam_file_for_test
    ~name:"satysfi-grcnum"
    ~version:"0.1"
    ()

let fontHash =
{|{
  "grcnum:grcnum-font":<"Single":{"src-dist":"grcnum/grcnum-font.ttf"}>
}|}

let makefile =
{|
PHONY: build-doc
build-doc:
	@echo "Target: build-doc"
|}

let files =
  [
    satysfi_grcnum_opam;
    "Satyristes", satyristes;
    "README.md", "@@README.md@@";
    "fonts.satysfi-hash", fontHash;
    "grcnum.satyh", "@@grcnum.satyh@@";
    "font.ttf", "@@font.ttf@@";
    "doc-grcnum.saty", "@@doc-grcnum.saty@@";
    "doc-example.saty", "@@doc-example.saty@@";
    "Makefile", makefile;
  ]

let env ~dest_dir:_ ~temp_dir : Satyrographos.Environment.t t =
  let open Shexp_process.Infix in
  let pkg_dir = FilePath.concat temp_dir "pkg" in
  let prepare_pkg =
    PrepareDist.empty pkg_dir
    >> prepare_files pkg_dir files
    >> stdout_to (FilePath.concat pkg_dir "Satyristes") (echo satyristes)
    >> stdout_to (FilePath.concat pkg_dir "README.md") (echo "@@README.md@@")
    >> stdout_to (FilePath.concat pkg_dir "fonts.satysfi-hash") (echo fontHash)
    >> stdout_to (FilePath.concat pkg_dir "grcnum.satyh") (echo "@@grcnum.satyh@@")
    >> stdout_to (FilePath.concat pkg_dir "font.ttf") (echo "@@font.ttf@@")
    >> stdout_to (FilePath.concat pkg_dir "doc-grcnum.saty") (echo "@@doc-grcnum.saty@@")
    >> stdout_to (FilePath.concat pkg_dir "doc-example.saty") (echo "@@doc-example.saty@@")
    >> stdout_to (FilePath.concat pkg_dir "Makefile") (echo makefile)
  in
  let empty_dist = FilePath.concat temp_dir "empty_dist" in
  let prepare_dist = PrepareDist.empty empty_dist in
  let opam_reg = FilePath.concat temp_dir "opam_reg" in
  let log_file = exec_log_file_path temp_dir in
  let prepare_opam_reg =
    PrepareOpamReg.(prepare opam_reg theanoFiles)
    >> PrepareOpamReg.(prepare opam_reg grcnumFiles)
    >> PrepareOpamReg.(prepare opam_reg classGreekFiles)
    >> PrepareOpamReg.(prepare opam_reg baseFiles)
  in
  let bin = FilePath.concat temp_dir "bin" in
  prepare_pkg
  >> prepare_dist
  >> prepare_opam_reg
  >> PrepareBin.prepare_bin bin log_file
  >>| read_env ~opam_reg ~dist_library_dir:empty_dist

let () =
  let verbose = false in
  let main env ~dest_dir:_ ~temp_dir ~outf =
    let names = Some ["example-doc"] in
    (* let dest_dir = FilePath.concat dest_dir "dest" in *)
    Satyrographos_command.Build.build_command
      ~outf
      ~verbose
      ~buildscript_path:(FilePath.concat temp_dir "pkg/Satyristes")
      ~build_dir:(FilePath.concat temp_dir "pkg/_build" |> Core.Option.some)
      ~env
      ~names
  in
  eval (test_install env main)
