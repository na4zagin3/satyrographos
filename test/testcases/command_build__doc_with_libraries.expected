Installing packages
------------------------------------------------------------
= Pin projects

= Build modules: (example-doc)

== Build module example-doc
== Install dependencies: (fonts-theano grcnum)
=== Build docs
Target: build-doc
================
Reading runtime dist: @@temp_dir@@/empty_dist
Reading opam libraries: (base class-greek fonts-theano grcnum)
Not gathering system fonts
Installing libraries: (dist fonts-theano grcnum)
Removing destination @@temp_dir@@/pkg/_build/satysfi/dist
Installation completed!

[1;33mCompatibility notice[0m for library fonts-theano:

  Fonts have been renamed.
  
    TheanoDidot -> fonts-theano:TheanoDidot
    TheanoModern -> fonts-theano:TheanoModern
    TheanoOldStyle -> fonts-theano:TheanoOldStyle

[1;33mCompatibility notice[0m for library grcnum:

  Packages have been renamed.
  
    grcnum.satyh -> grcnum/grcnum.satyh
------------------------------------------------------------
@@dest_dir@@
------------------------------------------------------------
------------------------------------------------------------
Command invoked:
opam pin add --no-action --yes satysfi-grcnum file://@@temp_dir@@/pkg
Command invoked:
opam reinstall --verbose --yes -- @@temp_dir@@/pkg
Command invoked:
opam install --yes satysfi-fonts-theano satysfi-grcnum
Command invoked:
satysfi --version
Command invoked:
satysfi -C @@temp_dir@@/pkg/_build/satysfi --no-default-config doc-example.saty -o doc-example-ja.pdf
doc-example.saty -> doc-example-ja.pdf
